packages/payload/src/fields/config/sanitize.ts

```diff
@@ -44,8 +53,41 @@ export const sanitizeFields = ({
     }
 
     // Make sure that the richText field has an editor
-    if (field.type === 'richText' && !field.editor && config.editor) {
-      field.editor = config.editor
+    if (field.type === 'richText') {
```

Comment:
All of the editor configuration logic should be done in a new file. These changes from L56 - L90 are hard to read.